src/test/java/net/snowflake/client/jdbc/ConnectionIT.java

```diff
@@ -543,7 +547,12 @@ public void testDifferentKeyLength() throws Exception {
       properties.put("privateKey", privateKey);
       try (Connection connection = DriverManager.getConnection(uri, properties);
           Statement statement = connection.createStatement()) {
-        statement.execute(String.format("alter user %s unset rsa_public_key", testUser));
+        assertFalse(connection.isClosed());
+      } finally {
+        try (Connection connection = getConnection();
```

Comment:
also I'm worried that getConnection here authenticates with a password accessed [here](https://github.com/snowflakedb/snowflake-jdbc/blob/e95bba2816769fc59bfb3f2e3424b449c4b9a323/src/test/java/net/snowflake/client/AbstractDriverIT.java#L300) instead of the private key so it doesn't really test the feature that it's supposed to test

Reply:
never mind, I didn't notice that the key is used for the assert so let's just add the use role in finally as mentioned in another comment